dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/notification/ProgramNotificationTemplateService.java

```diff
@@ -45,4 +48,27 @@
     void delete( ProgramNotificationTemplate programNotificationTemplate );
 
     List<ProgramNotificationTemplate> getProgramNotificationByTriggerType( NotificationTrigger trigger );
+
+    /**
+     * Method is used to check if {@link Program} is associated with any web
+     * hook notification template.
+     *
+     * @param program {@link Program} to check for association.
+     * @return true if {@link Program} is associated, otherwise false.
+     */
+    boolean isProgramLinkedToWebHookNotification( Program program );
+
+    /**
+     * Method is used to check if {@link ProgramStage} is associated with any
+     * web hook notification template.
+     *
+     * @param programStage {@link ProgramStage} to check for association.
+     * @return true if {@link ProgramStage} is associated, otherwise false.
+     */
+    boolean isProgramStageLinkedToWebHookNotification( ProgramStage programStage );
+
+    List<ProgramNotificationTemplate> getProgramLinkedToWebHookNotifications( Program program );
+
+    List<ProgramNotificationTemplate> getProgramStageLinkedToWebHookNotifications( ProgramStage programStage );
```

Comment:
All of these in the end just need the ID of the `Program` or `ProgramStage`. 
Hard to say for me if one always has these objects where these methods are used but I feel it would be more in the spirit of DHIS2 conventions to use UID and only ask for the UID as parameter here.
If I saw correctly the events do come with the UIDs for example.

Reply:
I was also thinking about the `getX` methods using UID parameters as they just use the instance to filter by ID which could equally well be lookup by UID or is there anything to be said against that?

Reply:
Out of curiosity (not an expert) - shouldn't the getters (`getProgramLinkedToWebHookNotifications` and `getProgramStageLinkedToWebHookNotifications`) also just take the program / programStage uid as a parameter, rather than the full program/programstage object....?

I guess this is what @jbee also asked.  All four method signatures here accept the full object, as do both the getters in `ProgramNotificationTemplateStore`, but both boolean `is` functions in `ProgramNotificationTemplateStore` accept Long ids.  All 8 implementations only use the id. I think we should at least be consistent and probably using the minimal id representation makes the most sense here?

Of course just let me know if I'm missing some context or Java magic here 👍

Reply:
@amcgee  `is` functions make use of uid(String) and id(Long). Uid for caching and id for fetching notification templates.

Reply:
id(Long) is vital to make fetching of templates more performant.